### PR TITLE
Revise description of when to send candidates. Fixes #211

### LINKED ISSUE #211: Update BUNDLE text to reflect BUNDLE draft 27
BUNDLE draft 27 (https://www.ietf.org/id/draft-ietf-mmusic-sdp-bundle-negotiation-27.txt) indicates that ICE attributes and candidates in re-offers (and with max-bundle, I'm guessing) should only be in the "m=" line associated with the offerer BUNDLE-tag.  In other words, the candidates and ICE parameters are not duplicated unnecessarily across m= sections. 

We should update JSEP to reflect this.

### PR DESCRIPTION
**Pull request by ekr**

### CODE REVIEW

**Comment by ekr** [comment]
First cut. Need to update the examples.

File: draft-ietf-rtcweb-jsep.xml [hunk 1]
```diff
@@ -461,7 +461,11 @@ setRemote(ANSWER) |                                   |
 
           <t>Upon receipt of trickled candidates, the receiving application
           will supply them to its ICE Agent. This triggers the ICE Agent to
-          start using the new remote candidates for connectivity checks.</t>
+          start using the new remote candidates for connectivity checks.
+          If BUNDLE is in use, each candidate is only supplied once, and
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: Probably need xref here for BUNDLE.

Suggest making new paragraph here that can be specific to BUNDLE - might be a better fit for the ICE candidate section, which is where the m= section association is introduced.

Suggestion for text clarification: "When BUNDLE is in use, candidates are only exchanged for and associated with the m= section referenced by the sender's BUNDLE-tag, as described in <blah>".

File: draft-ietf-rtcweb-jsep.xml [hunk 2]
```diff
@@ -1459,15 +1463,10 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
-          use the same ICE credentials and candidates as the m= section
-          into which it is being bundled. Note that this means
-          that for offers, any m= sections which are not bundle-only
-          MUST have unique ICE credentials and candidates, since it is
-          possible that the answerer will accept them without bundling
-          them.
+          <t>Each m= section, provided it is not marked as
+          bundle-only, MUST generate a unique set of ICE credentials and gather
+          its own unique set of ICE candidates. Bundle-only m= sections MUST NOT
+          have any ICE credentials and MUST NOT gather any candidates.
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: have->contain

File: draft-ietf-rtcweb-jsep.xml [hunk 3]
```diff
@@ -1709,19 +1708,25 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
 
               <t>Each "a=mid" line MUST stay the same.</t>
 
-              <t>Each "a=ice-ufrag" and "a=ice-pwd" line MUST stay the
+              <t>If the m= section is not bundled into another m= section
+              (i.e., it is either not bundled or it is associated with the
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: is associated with -> possesses (has)?

File: draft-ietf-rtcweb-jsep.xml [hunk 4]
```diff
@@ -1709,19 +1708,25 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
+              sender's BUNDLE-tag) each "a=ice-ufrag" and "a=ice-pwd" line MUST stay the
               same, unless the ICE configuration has changed (either changes to
               the supported STUN/TURN servers, or the ICE candidate policy), or
               the "IceRestart" option (<xref target="sec.icerestart"/>
-              was specified.
+              was specified. If the m= section is bundled into another
+              m= section, it MUST NOT have any ICE credentials.
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: have -> contain

File: draft-ietf-rtcweb-jsep.xml (hunk 2 above)
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: suggest adding ref to the appropriate section of bundle here

File: draft-ietf-rtcweb-jsep.xml [hunk 5]
```diff
@@ -1709,19 +1708,25 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
               been gathered during the most recent gathering phase
               (see <xref target="sec.ice-gather-overview"></xref>),
-              an "a=candidate" line MUST be added, as specified in <xref
-              target="RFC5245"></xref>, Section 4.3., paragraph 3. If
-              candidate gathering for the section has completed, an
+              an "a=candidate" line MUST be added, as defined in <xref
+              target="RFC5245"></xref>, Section 4.3., paragraph 3.
+              If the m= section is bundled into another m= section, any
+              "a=candidate" lines MUST be omitted.
+              If candidate gathering for the section has completed, an
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: Do we need to say anything about bundling here?

File: draft-ietf-rtcweb-jsep.xml [hunk 6]
```diff
@@ -2198,15 +2203,25 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
               type have yet been gathered, dummy values MUST be used, as
               described in the initial answer section above.</t>
 
-              <t>Each "a=ice-ufrag" and "a=ice-pwd" line MUST stay the same.
+              <t>If the m= section is not bundled into another m= section
+              (i.e., it is either not bundled or it is associated with the
+              sender's BUNDLE-tag) each "a=ice-ufrag" and "a=ice-pwd" line MUST stay the
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: see above

File: draft-ietf-rtcweb-jsep.xml [hunk 7]
```diff
@@ -2198,15 +2203,25 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
+              same, unless the ICE configuration has changed (either changes to
+              the supported STUN/TURN servers, or the ICE candidate policy), or
+              the "IceRestart" option (<xref target="sec.icerestart"/>
+              was specified. If the m= section is bundled into another
+              m= section, it MUST NOT have any ICE credentials.
               </t>
 
-              <t>Within each m= section, for each candidate that has
+              <t>If the m= section is not bundled into another m= section,
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: See above

**Comment by juberti** [comment]
Overall LG, a handful of comments

File: draft-ietf-rtcweb-jsep.xml (hunk 5 above)
Comment (ekr) [inline draft-ietf-rtcweb-jsep.xml]: Expand on this? The next sentence should make it clear

**Comment by fluffy** [comment]
LGTM

File: draft-ietf-rtcweb-jsep.xml (hunk 1 above)
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: Wdyt about the location of this text - "Suggest making new paragraph here that can be specific to BUNDLE - might be a better fit for the ICE candidate section, which is where the m= section association is introduced."

**Comment by juberti** [comment]
lgtm and merged. Will do a new PR to handle minor tweaks

**Pull request merged**